elasticapm/contrib/sanic/__init__.py

```diff
@@ -0,0 +1,168 @@
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+#  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+#  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+
+
+from __future__ import absolute_import
+
+import sys
+import typing as t
+
+from sanic import Sanic
+from sanic.request import Request
+from sanic.response import HTTPResponse
+
+from elasticapm import set_context as elastic_context
+from elasticapm import set_transaction_name, set_transaction_outcome, set_transaction_result
+from elasticapm.base import Client
+from elasticapm.conf import constants, setup_logging
+from elasticapm.contrib.asyncio.traces import set_context
+from elasticapm.contrib.sanic.utils import get_request_info, get_response_info, make_client
+from elasticapm.handlers.logging import LoggingHandler
+from elasticapm.instrumentation.control import instrument
+from elasticapm.utils.disttracing import TraceParent
+
+
+class ElasticAPM:
+    def __init__(
+        self,
+        app: Sanic,
+        client: t.Union[None, Client] = None,
+        client_cls: t.Type[Client] = Client,
+        log_level: int = 0,
+        config: t.Union[None, t.Dict[str, t.Any]] = None,
+        transaction_name_callback: t.Union[None, t.Callable[[Request], str]] = None,
+        **defaults,
+    ) -> None:
+        self._app = app  # type: Sanic
+        self._logging = log_level  # type: int
+        self._client_cls = client_cls  # type: type
+        self._client = client  # type: t.Union[None, Client]
+        self._logger = None
+        self._skip_headers = defaults.pop("skip_headers", [])  # type: t.List[str]
+        self._transaction_name_callback = transaction_name_callback  # type: t.Union[None, t.Callable[[Request], str]]
+        self._init_app(config=config, **defaults)
+
+    async def capture_exception(self, *args, **kwargs):
+        assert self._client, "capture_exception called before application configuration is initialized"
+        return self._client.capture_exception(*args, **kwargs)
+
+    async def capture_message(self, *args, **kwargs):
+        assert self._client, "capture_message called before application configuration is initialized"
+        return self._client.capture_message(*args, **kwargs)
+
+    # noinspection PyBroadException
+    def _init_app(self, config: t.Union[None, t.Dict[str, t.Any]], **defaults) -> None:
+        if not self._client:
+            cfg = config or self._app.config.get("ELASTIC_APM")
+            self._client = make_client(config=cfg, client_cls=self._client_cls, **defaults)
+
+        setup_logging(LoggingHandler(client=self._client, level=10))
```

Comment:
Do you mean for the logging level to be fixed here?

Reply:
Oh no! That was when I was testing some debug level info. Need to fix this.